frontend/controller/app/identity.js

```diff
@@ -462,24 +463,35 @@ export default (sbp('sbp/selectors/register', {
   // Unlike the login function, the wrapper for logging out is used using a
   // dedicated selector to allow it to be called from the login selector (if
   // error occurs)
-  'gi.app/identity/_private/logout': async function (errorState: ?Object) {
+  'gi.app/identity/_private/logout': async function (errorState: ?Object, wipeOut?: boolean) {
     try {
       const state = errorState || cloneDeep(sbp('state/vuex/state'))
       if (!state.loggedIn) return
 
       const cheloniaState = await sbp('gi.actions/identity/logout')
 
-      const { encryptionParams } = state.loggedIn
+      const { identityContractID, encryptionParams } = state.loggedIn
       if (encryptionParams) {
         // If we're logging out, save the current Chelonia state under the
         // `.cheloniaState` key. This will be used later when logging in
         // to restore both the Vuex and Chelonia states
-        state.cheloniaState = cheloniaState
+        if (!wipeOut) {
+          state.cheloniaState = cheloniaState
 
-        await sbp('state/vuex/save', true, state)
+          await sbp('state/vuex/save', true, state)
+        }
         await sbp('gi.db/settings/deleteStateEncryptionKey', encryptionParams)
         await sbp('appLogs/pauseCapture', { wipeOut: true }) // clear stored logs to prevent someone else accessing sensitve data
       }
+      // These should already be deleted, but we attempt to delete them again,
+      // just in case.
+      if (wipeOut) {
```

Comment:
Wjat does this comment mean, that it should "already be deleted"?

We don't delete the state permanently unless `wipeOut` is set... that least we're not supposed to, otherwise it would cause a long login each time.

Reply:
Yes, we do delete all of those keys when logging out. It doesn't cause a long login each time because the state is loaded from the encrypted store.